tensorboard/backend/event_processing/suffix_based_directory_watcher.py

```diff
@@ -0,0 +1,272 @@
+
+    Yields:
+      All values that have not been yielded yet.
+    """
+    while True:
+      # Used to determine whether we should keep obtaining new directory
+      # listings. We only do so if we picked up new events to read.
+      at_least_1_event_read = False
+
+      # Sort event file names by timestamp (at which it was written) and then
+      # suffix.
+      paths = [path
+               for path in io_wrapper.ListDirectoryAbsolute(self._directory)
+               if self._path_filter(path) and path not in self._closed_paths]
+      paths.sort()
+
+      # Map from suffix to the index of the last events file with that suffix.
+      # Used later to determine whether to close an events file - we close the
+      # file if a subsequent file with the same suffix exists.
+      suffix_to_last_file_index = {}
+      for (i, path) in enumerate(paths):
+        suffix = self._ParseSuffix(path)
+        if not suffix:
+          continue
+        suffix_to_last_file_index[suffix] = i
+
+      while True:
+        # Whether any event has been read from any of the files within the
+        # current file listing.
+        at_least_1_event_read_for_current_paths = False
+
+        for (i, path) in enumerate(paths):
+          if path in self._closed_paths:
+            # Never again read from closed files.
+            continue
+
+          # Parse the suffix from the name of the events file.
+          suffix = self._ParseSuffix(path)
+          if not suffix:
+            tf.logging.info(
+                'File %r does not match events file naming pattern', path)
+            continue
+
+          if suffix not in self.suffix_to_head:
+            # Read this file. This file has not been read from before.
+            head = self._CreateNewHead(suffix, path)
+          elif path < self.suffix_to_head[suffix].path:
+            tf.logging.error(
+                ('Events file %r is out of order since it is ordered behind '
+                 '%r. It will not be read.'),
+                path,
+                self.suffix_to_head[suffix].path)
+            continue
+          else:
+            # Keep reading from the head of the current suffix. It still has
+            # more events to read.
+            head = self.suffix_to_head[suffix]
+
+          # Yield all remaining events for this suffix.
+          last_event = None
+          for event in head.loader.Load():
+            last_event = event
+            yield event
+
+          if last_event:
+            at_least_1_event_read_for_current_paths = True
+            at_least_1_event_read = True
+            head.last_event_time = last_event.wall_time
+
+          # Close this file if either at least 1 events file (with the same
+          # suffix) comes after this events file or its last recorded event
+          # timestamp is old.
+          if (suffix_to_last_file_index[suffix] > i or (
+                head.last_event_time is not None and
```

Comment:
Indent should only be +4 spaces here, not +6.  Ditto for line below.

Reply:
Done!